### PR TITLE
MorganAJ-CodeReview_2

### PR DESCRIPTION
**Pull request by ajmorgs**
For the most part, javadoc adequately describes functions and classes. Unit tests have been created.

They correctly use 'val' for Immutable variables and 'var' for mutable ones.

Interface IPictureDAO is used correctly.

Nulls are handled correctly.

  Changed val view = binding.root
        return view
		to 
		return binding.root as variable is unnecessary
		
Remove commented and unused function getPictureByName() from interface IPictureDAO

Renamed button2 to buttonCreateNew in main_fragment.xml in accordance with Clean Coding guidelines

https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29

### CODE REVIEW

File: app/src/main/java/com/example/puzzlepic/dao/IPictureDAO.kt [hunk 1]
```diff
@@ -11,7 +11,4 @@ interface IPictureDAO {
     @GET("/photos?client_id=gNYmyVdZ2_qHp3L4i4yzrulyHU70MdvnmsFFyB_5y38&per_page=100")
     fun getRandomImage(): Call<ArrayList<Picture>>
 
-    //
-    //@GET("")
-    //fun getPictureByName(pictureName:String) : Call<ArrayList<Picture>> //going to return a specific picture.
```
Comment (discospiff) [inline app/src/main/java/com/example/puzzlepic/dao/IPictureDAO.kt:16]: ? 
What is the rationale for this change?

 Committing commented out code is considered bad practice in version control, because version control offers better alternatives:
1) Use a feature branch for experimental features.
2) Delete what is not needed.  You can always compare with history to see what used to be there.
Commented out code is considered clutter in a version control system.

File: app/src/main/res/layout/main_fragment.xml [hunk 2]
```diff
@@ -317,7 +317,7 @@
         app:layout_constraintStart_toStartOf="parent">
 
         <Button
-            android:id="@+id/button2"
+            android:id="@+id/buttonCreateNew"
```
Comment (discospiff) [inline app/src/main/res/layout/main_fragment.xml:320]: This breaks the build.  Check CircleCI output.

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 3]
```diff
@@ -61,11 +61,12 @@ class MainFragment : Fragment() {
...
-                              savedInstanceState: Bundle?): View {
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
         _binding = MainFragmentBinding.inflate(inflater, container, false)
-        val view = binding.root
-        return view
+        return binding.root
     }
```
Comment (discospiff) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:70]: For reformatting... 
1) A caution.  Significant reformats can make future commit diffs difficult, as git has to deal with significant reordering of the logic.
2) The IDE can do this automatically, and you can provide a template for the IDE to use.  Ensure all developers are using this same template, or the one that comes by default with the IDE, and have them reformat before committing.  This ameliorates issue 1, above.

**Comment by discospiff** [comment]
I don't recommend merging this branch, as:
1) It's not clear on how it reduces technical debt.
2) Most importantly, it breaks the build.  Check CircleCI output for details.

**Comment by erswagner7** [comment]
Decided not to merge code review suggestions based off failed circleCI tests
